x-pack/metricbeat/module/openai/usage/usage.go

```diff
@@ -172,12 +153,16 @@ func (m *MetricSet) fetchSingleDay(dateStr, apiKey string, httpClient *RLHTTPCli
 
 	resp, err := httpClient.Do(req)
 	if err != nil {
+		if err, ok := err.(net.Error); ok && err.Timeout() {
+			return ErrHTTPClientTimeout
```

Comment:
If you have access to the timeout, it's often useful to add it to the error message

Reply:
Done!